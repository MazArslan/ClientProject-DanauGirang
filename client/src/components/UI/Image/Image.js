import React from "react";
import propTypes from "prop-types";
import { Container, Row, Col } from "reactstrap";
import "./Image.css";

/**
 * Image component
 * Renders an image
 * @param { filePath, id } props
 */
const image = props => {
  // eslint-disable-next-line no-unused-vars
  const { filePath, id } = props;
  return (
    <Container className="image-component">
      <Row key={image._id}>
        <Col className="text-center">
          <img
            src={`/api/images/cameraTrap/${filePath}`}
            alt={`${id}`}
            style={{ maxHeight: "600px", maxWidth: "720px" }}
          />
        </Col>
      </Row>
    </Container>
  );
};

image.propTypes = {
  filePath: propTypes.string.isRequired,
  id: propTypes.string
};

export default image;
