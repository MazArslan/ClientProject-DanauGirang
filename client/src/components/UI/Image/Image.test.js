import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import Image from "./Image";

Enzyme.configure({ adapter: new Adapter() });

describe("<ImageThumbnail/>", () => {
  const props = {
    id: "Unique id string",
    filePath: "5e123ty34u35567h2g/0001.png"
  };
  const container = shallow(<Image {...props} />);

  describe("component rendering", () => {
    it("should render correctly", () => {
      expect(container.exists()).toBe(true);
      expect(toJson(container)).toMatchSnapshot();
    });

    it("should render an image", () => {
      const imageString = "/api/images/cameraTrap/5e123ty34u35567h2g/0001.png";
      expect(container.find("img").props().src).toEqual(imageString);
    });
  });
});
