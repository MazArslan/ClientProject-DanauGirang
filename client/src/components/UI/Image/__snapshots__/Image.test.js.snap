// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageThumbnail/> component rendering should render correctly 1`] = `
<Container
  className="image-component"
  tag="div"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="text-center"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <img
        alt="Unique id string"
        src="/api/images/cameraTrap/5e123ty34u35567h2g/0001.png"
        style={
          Object {
            "maxHeight": "600px",
            "maxWidth": "720px",
          }
        }
      />
    </Col>
  </Row>
</Container>
`;
